cloudinit/config/cc_growpart.py

```diff
@@ -210,11 +210,35 @@ def get_size(filename):
 
 def device_part_info(devpath):
     # convert an entry in /dev/ to parent disk and partition number
+    myenv = os.environ.copy()
+    myenv['LANG'] = 'C'
 
     # input of /dev/vdb or /dev/disk/by-label/foo
     # rpath is hopefully a real-ish path in /dev (vda, sdb..)
     rpath = os.path.realpath(devpath)
 
+    is_lvm = True if re.search('.*dm.*', rpath) else False
+    if is_lvm:
+        try:
+            (out, _err) = subp.subp(["lvm", "lvs", devpath, "-o", "vgname",
+                                     "--noheadings"], env=myenv)
+            vgname = out.strip()
+        except subp.ProcessExecutionError as e:
+            if e.exit_code != 0:
+                util.logexc(LOG, "Failed: can't get Volume Group information "
+                                 "from %s", devpath)
+                raise ResizeFailedException(e) from e
+
+        try:
+            (out, _err) = subp.subp(["lvm", "vgs", vgname.strip(), "-o",
+                                    "pvname", "--noheadings"], env=myenv)
+            rpath = out.strip()
```

Comment:
a VG can have more than one PV. if it does, this would give you a string like '/dev/vdb1\n/dev/vdb2'